### ISSUE TITLE
'Missing Semicolon' error when using definite assignment assertion modifier (aka damnit operator) in Typescript

### ISSUE THREAD

**Issue by PaulGarnerUk** [palantir/tslint#3811]
### Bug Report

- __TSLint version__: 5.9.1 
- __TypeScript version__: 2.7.2
- __Running TSLint via__: Visual Studio

#### TypeScript code being linted

```ts
class SomeClass
{
	SomeProperty!: number;
}
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"semicolon": [ true, "always" ]
	}
}
```

#### Actual behavior
Linter errors 'Missing semicolon'

#### Expected behavior
No error. Seems the linter is not correctly parsing the new syntax introduced in Typescript 2.7.x which comes into play when using strictPropertyInitialization (and strictNullChecks). [Described here under the section 'Strict Class Initialization'](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html)

**Comment by JoshuaKGoldberg** [comment]
@PaulGarnerUk TSLint uses whatever version of TypeScript is resolved when it `require("typescript")`s it at runtime. If you're seeing this kind of error, make sure the version of TypeScript you have installed alongside TSLint is up-to-date.
